### PR TITLE
Improve RSAPSS documentation

### PR DESCRIPTION
**Pull request by ma-ilsi**
## Proposed changes

The documentation for RSAPSS receives additional clarity:

- Return value of `new_rsapss_load_pkey` and `new_rsapss_load_skey` explains that NULL is returned on failure.
- Supported hash algorithms are listed in every applicable function and rendered in proper bullet point form, now. The use of `*` to create Doxygen lists can cause trouble when also using markdown for documentation. Doxygen itself admits [special care](https://www.doxygen.nl/manual/markdown.html#mddox_stars) needs to be taken. If we simply replace the `*` with `-` to make a list, it works without extra worries.
- Endianness/Byte-order addressed for key part parameters.
- Additionally, the disjoint memory pre-condition for applicable parameters _should_ be mentioned somewhere. Although it is non-sensible in most functions for a client to attempt memory overlapping, verification guarantees are off the table if it is overlapped, and so a note on that should appear generally somewhere in docs. This PR does not address this issue.

These changes resolve the issues expressed in cryspen/hacl-packages#345 & cryspen/hacl-packages#346. 

## Types of changes

What types of changes does your code introduce to HACL*?
_Put an `x` in the boxes that apply_

- [ ] Proof maintenance (non-breaking change which fixes a proof regression)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New algorithm or feature (non-breaking change which adds functionality)
- [ ] Improved performance (fix or feature that would improve performance of some algorithm on some platform)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CODE_OF_CONDUCT.md](https://github.com/hacl-star/hacl-star/blob/main/CODE_OF_CONDUCT.md) doc
- [x] I have read and agree to submit my changes under the [LICENSE](https://github.com/hacl-star/hacl-star/blob/main/LICENSE)
- [x] I have added necessary documentation (if appropriate)
- [ ] I have edited [CHANGES.md](https://github.com/hacl-star/hacl-star/blob/main/CHANGES.md) (if appropriate)

### CODE REVIEW

**Review by polubelova — APPROVED** [review]
Thanks!

**Review by msprotz — APPROVED** [review]

**Comment by msprotz** [comment]
```
* Error 168 at /build/source/code/rsapss/Hacl.RSAPSS.fst(107,1-107,1):
hacl-star>   - Syntax error
```

File: code/rsapss/Hacl.RSAPSS.fst [hunk 1]
```diff
@@ -90,10 +93,10 @@ let rsapss_verify a modBits eBits pkey saltLen sgntLen sgnt msgLen msg =
 
 @param modBits Count of bits in modulus (`n`).
 @param eBits Count of bits in `e` value.
-@param nb Pointer to `ceil(modBits / 8)` bytes where the modulus (`n`) is read from.
-@param eb Pointer to `ceil(modBits / 8)` bytes where the `e` value is read from.
+@param nb Pointer to `ceil(modBits / 8)` bytes where the modulus (`n`), in big-endian byte order, is read from.
+@param eb Pointer to `ceil(modBits / 8)` bytes where the `e` value, in big-endian byte order, is read from.
 
-@return Returns an allocated public key. Note: caller must take care to `free()` the created key."]
+@return Returns an allocated public key upon success, otherwise, `NULL` if key part arguments are invalid or memory allocation fails. Note: caller must take care to `free()` the created key.]
```
Comment (msprotz) [inline code/rsapss/Hacl.RSAPSS.fst:99]: missing ending " here

File: code/rsapss/Hacl.RSAPSS.fst [hunk 2]
```diff
@@ -104,25 +107,28 @@ let new_rsapss_load_pkey modBits r eBits nb eb =
...
 @param dBits Count of bits in `d` value.
-@param nb Pointer to `ceil(modBits / 8)` bytes where the modulus (`n`) is read from.
-@param eb Pointer to `ceil(modBits / 8)` bytes where the `e` value is read from.
-@param db Pointer to `ceil(modBits / 8)` bytes where the `d` value is read from.
+@param nb Pointer to `ceil(modBits / 8)` bytes where the modulus (`n`), in big-endian byte order, is read from.
+@param eb Pointer to `ceil(modBits / 8)` bytes where the `e` value, in big-endian byte order, is read from.
+@param db Pointer to `ceil(modBits / 8)` bytes where the `d` value, in big-endian byte order, is read from.
 
-@return Returns an allocated secret key. Note: caller must take care to `free()` the created key."]
+@return Returns an allocated secret key upon success, otherwise, `NULL` if key part arguments are invalid or memory allocation fails. Note: caller must take care to `free()` the created key.]
```
Comment (msprotz) [inline code/rsapss/Hacl.RSAPSS.fst:114]: same here

**Comment by msprotz** [comment]
if you can fix the missing closing quotes, then I'm happy to merge this... thanks

**Comment by ma-ilsi** [comment]
Updated

**Comment by msprotz** [comment]
Excellent, thanks so much! Great to see this kind of improvements.

**Pull request merged**